File: Baekjoon/구현/10811.py
# https://www.acmicpc.net/problem/10811

import sys
input = sys.stdin.readline

n, m = map(int, input().split())
arr = [i for i in range(1, n + 1)]

for _ in range(m):
    i, j = map(int, input().split())
    arr = arr[:i - 1] + arr[i - 1:j][::-1] + arr[j:]
    
print(*arr)

File: Baekjoon/구현/10811.cpp
// https://www.acmicpc.net/problem/10811

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        a[i] = i + 1;
    }
    while (m--) {
        int i, j;
        cin >> i >> j;
        reverse(a.begin() + i - 1, a.begin() + j);
    }
    for (int i = 0; i < n; i++) {
        cout << a[i] << ' ';
    }
    return 0;
}